contentreview-impl/impl/src/java/org/sakaiproject/contentreview/impl/turnitin/TurnitinReviewServiceImpl.java

```diff
@@ -3425,14 +3344,49 @@ private String switchLTIError(int result, String method){
 				return method + ": generic LTI error";
 		}
 	}
-	
-	private void processError(ContentReviewItem item, Long status, String error){//TODO resto de llamadas
-		item.setStatus(status);
-		if(error != null){
+
+	private void processError( ContentReviewItem item, Long status, String error, Integer errorCode )
+	{
+		if( status == null )
+		{
+			IllegalArgumentException ex = new IllegalArgumentException( "Status is null; you must supply a valid status to update when calling processError()" );
+			log.warn( ex );
```

Comment:
When code logs when the exception is generated I normally end up seeing 2 stack traces in the logs as the calling code logs it as well. If the calling code is just throwing away an exception when it should be logging it that it's better to make the fix there are you'll start catching other ignored exceptions.

But 👍  for adding argument validation to methods, my normal rule of thumb is that validation is really important on external facing methods where you can't be sure that the caller is doing the right thing.